src/controllers/dynakube/version/image_hash.go

```diff
@@ -2,80 +2,92 @@ package version
 
 import (
 	"context"
+	"crypto/tls"
+	"crypto/x509"
 	"fmt"
-
-	"github.com/Dynatrace/dynatrace-operator/src/dockerconfig"
-	"github.com/containers/image/v5/image"
-	"github.com/containers/image/v5/manifest"
-	"github.com/containers/image/v5/transports/alltransports"
-	"github.com/containers/image/v5/types"
-	"github.com/opencontainers/go-digest"
-	"github.com/pkg/errors"
-)
-
-const (
-	// VersionLabel is the name of the label used on ActiveGate-provided images.
-	VersionLabel = "com.dynatrace.build-version"
+	"net/http"
+	"net/url"
+
+	dynatracev1beta1 "github.com/Dynatrace/dynatrace-operator/src/api/v1beta1/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/src/dockerkeychain"
+	"github.com/Dynatrace/dynatrace-operator/src/registry"
+	"github.com/spf13/afero"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-type ImageVersion struct {
-	Version string
-	Digest  digest.Digest
-}
-
 // ImageVersionFunc can fetch image information from img
-type ImageVersionFunc func(ctx context.Context, imageName string, dockerConfig *dockerconfig.DockerConfig) (ImageVersion, error)
+type ImageVersionFunc func(
+	ctx context.Context,
+	registryClient registry.ImageGetter,
+	imageName string,
+	registryAuthPath string,
+	dynakube *dynatracev1beta1.DynaKube,
+	apiReader client.Reader,
+) (
+	registry.ImageVersion,
+	error,
+)
 
 var _ ImageVersionFunc = GetImageVersion
 
 // GetImageVersion fetches image information for imageName
-func GetImageVersion(ctx context.Context, imageName string, dockerConfig *dockerconfig.DockerConfig) (ImageVersion, error) {
-	transportImageName := fmt.Sprintf("docker://%s", imageName)
-
-	imageReference, err := alltransports.ParseImageName(transportImageName)
-	if err != nil {
-		return ImageVersion{}, errors.WithStack(err)
+func GetImageVersion( //nolint:revive // argument-limit
+	ctx context.Context,
+	registryClient registry.ImageGetter,
+	imageName string,
+	registryAuthPath string,
+	dynakube *dynatracev1beta1.DynaKube,
+	apiReader client.Reader,
```

Comment:
nitpick:
I would have a different order of args
`(ctx, apiReader, registryClient,  dynakube, imageName, registryAuthPath)` 
I like to have the "simple types" at the end
- So general ordering I  ⚠️ **try**⚠️  to follow
    1.  Convention (ctx first)
    2. Clients + stuff with interfaces
    3. Data structs
    4. Simple/base types
    
(don't feel the need to refactor everything, just maybe look at the new additions and try to be consistent)

Reply:
Thank you for the explanation :+1::+1::+1:

Reply:
Done [here](https://github.com/Dynatrace/dynatrace-operator/pull/1998/commits/c9043326416b8b00f85335b458127240cfeb820a)